transport/tchannel/outbound_test.go

```diff
@@ -40,216 +40,160 @@ import (
-			ctx,
-			&transport.Request{
-				Caller:    "caller",
-				Service:   "service",
-				Encoding:  raw.Encoding,
-				Procedure: "hello",
-				Headers:   tt.headers,
-				Body:      bytes.NewReader([]byte("world")),
-			},
-		)
-		if tt.wantError != "" {
-			if assert.Error(t, err, "expected error") {
-				assert.Contains(t, err.Error(), tt.wantError)
-			}
-		} else {
-			if assert.NoError(t, err, "call failed") {
-				defer res.Body.Close()
-			}
-		}
-	}
-}
-
-func TestCallSuccess(t *testing.T) {
-	const (
-		headerKey = "foo-BAR-BaZ"
-		headerVal = "FooBarBaz"
-	)
-	tests := []struct {
-		msg                   string
-		options               []TransportOption
-		headerVal             []byte
-		withServiceRespHeader bool
-	}{
-		{
-			msg:     "exactCaseHeader options on",
-			options: []TransportOption{ServiceName("caller"), OriginalHeaders()},
-			headerVal: []byte{
-				0x00, 0x01,
-				0x00, 0x0b, 'f', 'o', 'o', '-', 'B', 'A', 'R', '-', 'B', 'a', 'Z',
-				0x00, 0x09, 'F', 'o', 'o', 'B', 'a', 'r', 'B', 'a', 'z',
+			wantHeaders: map[string]string{
+				"foo-BAR-BaZ": "PiE",
+				"foo-bar":     "LEMON",
+				"BAR-BAZ":     "orange",
 			},
+			originalHeaders: true,
 		},
 		{
-			msg:     "exactCaseHeader options off",
-			options: []TransportOption{ServiceName("caller")},
-			headerVal: []byte{
-				0x00, 0x01,
-				0x00, 0x0b, 'f', 'o', 'o', '-', 'b', 'a', 'r', '-', 'b', 'a', 'z',
-				0x00, 0x09, 'F', 'o', 'o', 'B', 'a', 'r', 'B', 'a', 'z',
+			name: "exactCaseHeader options off",
+			giveHeaders: map[string]string{
+				"foo-BAR-BaZ": "PiE",
+				"foo-bar":     "LEMON",
+				"BAR-BAZ":     "orange",
 			},
-		},
-		{
-			msg:     "exactCaseHeader options off",
-			options: []TransportOption{ServiceName("caller")},
-			headerVal: []byte{
-				0x00, 0x01,
-				0x00, 0x0b, 'f', 'o', 'o', '-', 'b', 'a', 'r', '-', 'b', 'a', 'z',
-				0x00, 0x09, 'F', 'o', 'o', 'B', 'a', 'r', 'B', 'a', 'z',
+			wantHeaders: map[string]string{
+				"foo-bar-baz": "PiE",
+				"foo-bar":     "LEMON",
+				"bar-baz":     "orange",
 			},
-			withServiceRespHeader: true,
 		},
 	}
 
 	for _, tt := range tests {
-		t.Run(tt.msg, func(t *testing.T) {
+		t.Run(tt.name, func(t *testing.T) {
 			server := testutils.NewServer(t, nil)
 			defer server.Close()
 			serverHostPort := server.PeerInfo().HostPort
 
 			server.GetSubChannel("service").SetHandler(tchannel.HandlerFunc(
 				func(ctx context.Context, call *tchannel.InboundCall) {
-					assert.Equal(t, "caller", call.CallerName())
-					assert.Equal(t, "service", call.ServiceName())
-					assert.Equal(t, tchannel.Raw, call.Format())
-					assert.Equal(t, "hello", call.MethodString())
-					headers, body, err := readArgs(call)
-					if assert.NoError(t, err, "failed to read request") {
-						assert.Equal(t, tt.headerVal, headers)
-						assert.Equal(t, []byte("world"), body)
+					headers, err := readHeaders(tchannel.Raw, call.Arg2Reader)
+					if !assert.NoError(t, err, "failed to read request") {
```

Comment:
This is an issue from the original implementation but we can fix it now: We
should track whether the handler is actually called.

```
called := false
...
    func(ctx context.Context, call *tchannel.InboundCall) {
        called = true
    }
```

Then at the bottom after the client call,

```
assert.True(t, called, "handler was never called by client")
```